src/repr/src/row.rs

```diff
@@ -1102,6 +1115,92 @@ impl Row {
     }
 }
 
+/// The type returned by [Row::unpack_owned]
+pub struct OwnedDatums(Yoke<DatumVecYoke<'static>, Vec<u8>>);
```

Comment:
Would it be better to use a `Box<[u8]>` instead of a `Vec<u8>`? Saves a word IIUC.

Reply:
That's what I originally typed out but then `OwnedDatums::into_row` can't turn the `Box<[u8]>` back into a `Row` without reallocation so I thought it's better to keep one extra word to avoid the reallocation

Reply:
Wait, other way around, right? I thought `Vec<u8>` -> `Box<[u8]>` required reallocation (via `Vec::shrink_to_fit`), but `Box<[u8]>` -> `Vec<u8>` did not. Your point stands though.

Reply:
`Vec::shrink_to_fit` eventually calls into the `Allocator::shrink` method which may or may not re-allocate. I was confused about this too, which is why I chased down the method calls in `std`

Reply:
Right, but isn't `Box<[u8]>` -> `Vec<u8>` always reallocation free?

Reply:
Oh! Yes you're right, my original point was wrong.